internal/verifier/change_stream.go

```diff
@@ -352,3 +376,30 @@ func getClusterTimeFromSession(sess mongo.Session) (primitive.Timestamp, error)
 
 	return ctStruct.ClusterTime.ClusterTime, nil
 }
+
+func (verifier *Verifier) flushAllBufferedChangeEventRechecks(ctx context.Context) error {
+	for namespace, ids := range verifier.changeEventRecheckBuf {
+		if len(ids) == 0 {
+			return nil
+		}
+
+		// We don't know the document sizes for documents for all change events,
+		// so just be conservative and assume they are maximum size.
+		//
+		// Note that this prevents us from being able to report a meaningful
+		// total data size for noninitial generations in the log.
+		dataSizes := make([]int, len(ids))
+		for i := 0; i < len(ids); i++ {
+			dataSizes[i] = maxBSONObjSize
+		}
+
+		dbName, collName := SplitNamespace(namespace)
+		if err := verifier.insertRecheckDocs(ctx, dbName, collName, ids, dataSizes); err != nil {
```

Comment:
This approach will cause us to block between each namespace’s events, I think.

Could we instead persist all of the events for all namespaces at once?